pikaraoke/karaoke.py

```diff
@@ -189,18 +189,30 @@ def __init__(
 
         self.generate_qr_code()
 
-    # Other ip-getting methods are unreliable and sometimes return 127.0.0.1
-    # https://stackoverflow.com/a/28950776
     def get_ip(self):
-        s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
-        try:
-            # doesn't even have to be reachable
-            s.connect(("10.255.255.255", 1))
-            IP = s.getsockname()[0]
-        except Exception:
-            IP = "127.0.0.1"
-        finally:
-            s.close()
+        # python socket.connect will not work on android, access denied. Workaround: use ifconfig which is installed to termux by default, iirc.
+        if "Android" in self.platform:
```

Comment:
why not this more precise statement: `if (self.platform = "Android")`  (or `if (self.platform = "android")` if you follow my comment on making this lowercase)